packages/cozy-client/src/StackLink.spec.js

```diff
@@ -46,6 +46,22 @@ describe('StackLink', () => {
       expect(stackClient.collection().all).toHaveBeenCalled()
       expect(stackClient.collection).toHaveBeenCalledWith('io.cozy.todos')
     })
+
+    it('should use find if a partialFilter is given', async () => {
+      const query = Q('io.cozy.todos').partialIndex({ trashed: false })
+      stackClient.collection().find.mockReset()
+      await link.request(query)
+      expect(stackClient.collection().find).toHaveBeenCalled()
+      expect(stackClient.collection).toHaveBeenCalledWith('io.cozy.todos')
+    })
+
+    it('should use find if fields are given', async () => {
+      const query = Q('io.cozy.todos').select(['trashed'])
+      stackClient.collection().find.mockReset()
+      await link.request(query)
+      expect(stackClient.collection().find).toHaveBeenCalled()
+      expect(stackClient.collection).toHaveBeenCalledWith('io.cozy.todos')
+    })
```

Comment:
It might be worth adding a test for `all`
```
it('should use all when the whole doctype is queried',  async () => {
    const query = Q('io.cozy.todos')
    stackClient.collection().all.mockReset()
    await link.request(query)
    expect(stackClient.collection().all).toHaveBeenCalled()
    expect(stackClient.collection).toHaveBeenCalledWith('io.cozy.todos')
  })
 ```

Reply:
https://github.com/cozy/cozy-client/blob/4cdda31aa62a39e569defa737aef0bc33aa4901c/packages/cozy-client/src/StackLink.spec.js#L42-L48

already there ;)

Reply:
Uh, my bad

Reply:
Though, next time, I'll rename the test! it's not relevant ATM.